client/src/components/app.jsx

```diff
@@ -9,23 +9,29 @@ import {
 import './common/fontAwesomeIcons';
 
 import Banner from './common/banner.jsx';
-import ProductReviews from './productReviews/productReviews.jsx';
-import RelatedProducts from './relatedProducts/RelatedProducts.jsx';
-import Outfit from './relatedProducts/Outfit.jsx';
-import OverviewContainer from './productOverview/OverviewContainer.jsx';
+import Wrapper from './Wrapper.jsx';
 
-const App = () => (
-  <Router>
-    <Banner />
-    <Switch>
-      <Route exact path='/product/:productId'>
-        <OverviewContainer />
-        <RelatedProducts />
-        <Outfit />
-        <ProductReviews />
-      </Route>
-    </Switch>
-  </Router>
-);
+class App extends React.Component {
+  // eslint-disable-next-line no-useless-constructor
+  constructor(props) {
+    super(props);
+  }
+
+  render() {
+    return (
+      <Router>
+        <Banner />
+        <Switch>
+          <Route exact={false} path='/product/:productId' component={Wrapper} {...this.props}>
```

Comment:
Hey what does the `{...this.props} `do?

Reply:
it copies props (like history and location) and passes them into the wrapper component. For example, if you open react dev tools, click on components, click on the wrapper component, on the right, it should list the props.  As you click on the nested components, you will see that those that continue to have history passed in (like related products) will show history in the props, but children that don't have it passed down from wrapper won't have history listed in props.